tla-bmcmt/src/main/scala/at/forsyte/apalache/tla/bmcmt/rules2/SetCupStratifiedRule.scala

```diff
@@ -0,0 +1,56 @@
+package at.forsyte.apalache.tla.bmcmt.rules2
+
+import at.forsyte.apalache.tla.bmcmt._
+import at.forsyte.apalache.tla.bmcmt.arena.PtrUtil
+import at.forsyte.apalache.tla.bmcmt.types.CellT
+import at.forsyte.apalache.tla.lir.oper.TlaSetOper
+import at.forsyte.apalache.tla.lir.{OperEx, TlaEx}
+
+/**
+ * Rewrites X \cup Y, that is, a union of two sets (not UNION). In the first encoding, we used a linear number of `in`
+ * queries. However, this happens to be unsound, and we need a quadratic number of queries.
```

Comment:
What is the "first encoding"?

Reply:
On reading further, I doubt that we need this historical account?
Unless it somehow relates to the current implementation?

Reply:
IIRC, this is just left over from `SetCupRule`, which I duplicated and changed, but I didn't remove any comments attached to it.